lib/protobuf/json.ex

```diff
@@ -166,4 +186,93 @@ defmodule Protobuf.JSON do
   catch
     error -> {:error, EncodeError.new(error)}
   end
+
+  @doc """
+  Decodes a JSON `iodata` into a `module` Protobuf struct.
+
+  Similar to `decode!/2` except it will unwrap the error tuple and raise in case of errors.
+
+  ## Examples
+
+      iex> Protobuf.JSON.decode!("{}", Car)
+      %Car{color: :GREEN, top_speed: 0.0}
+
+      iex> ~S|{"color":"RED"}| |> Protobuf.JSON.decode!(Car)
+      %Car{color: :RED, top_speed: 0.0}
+
+      iex> ~S|{"color":"GREEN","topSpeed":80.0}| |> Protobuf.JSON.decode!(Car)
+      %Car{color: :GREEN, top_speed: 80.0}
+
+  """
+  @spec decode!(iodata, module) :: struct | no_return
+  def decode!(iodata, module) do
+    case decode(iodata, module) do
+      {:ok, json} -> json
+      {:error, error} -> raise error
+    end
+  end
+
+  @doc """
+  Decodes a JSON `iodata` into a `module` Protobuf struct.
+
+  ## Examples
+
+  Given these proto modules:
+
+      syntax = "proto3";
+
+      message Car {
+        enum Color {
+          GREEN = 0;
+          RED = 1;
+        }
+
+        Color color = 1;
+        float top_speed = 2;
+      }
+
+  You can build their structs from JSON like so:
+
+      iex> Protobuf.JSON.decode("{}", Car)
+      {:ok, %Car{color: :GREEN, top_speed: 0.0}}
+
+      iex> ~S|{"color":"RED"}| |> Protobuf.JSON.decode(Car)
+      {:ok, %Car{color: :RED, top_speed: 0.0}}
+
+      iex> ~S|{"color":"GREEN","topSpeed":80.0}| |> Protobuf.JSON.decode(Car)
+      {:ok, %Car{color: :GREEN, top_speed: 80.0}}
+  """
+  @spec decode(iodata, module) :: {:ok, struct} | {:error, DecodeError.t() | Exception.t()}
+  def decode(iodata, module) do
+    if Code.ensure_loaded?(Jason) do
+      with {:ok, json_data} <- Jason.decode(iodata),
```

Comment:
Just noticed this triggers a compilation warning on the `bench` project, which doesn't have `jason` in its deps.

```
==> protobuf
Compiling 33 files (.ex)
warning: Jason.decode/1 is undefined (module Jason is not available or is yet to be defined)
  lib/protobuf/json.ex:248: Protobuf.JSON.decode/2

warning: Jason.encode/1 is undefined (module Jason is not available or is yet to be defined)
  lib/protobuf/json.ex:161: Protobuf.JSON.encode/2
```

We'd probably need to conceal it in a variable to avoid it. I'll take a look.

Reply:
Yeah this is going to work:

```elixir
jason = Jason

if Code.ensure_loaded?(jason) do
  jason.decode(iodata)
```

:)

Reply:
Fixed with a private helper 👍